### ISSUE TITLE
McFly clobbers `$?` return status of interactive commands.

### ISSUE THREAD

**Issue by gwk** [cantino/mcfly#15]
For example:
```
$ false
$ echo $?
```
This should print 1, but prints 0 when mcfly is loaded. This breaks custom bash prompts that e.g. set prompt color based on `$?`.

**Comment by cantino** [comment]
Wait, it's working fine for me with McFly loaded.

```bash
~/projects/rust/mcfly (master)$ false
~/projects/rust/mcfly (master)$ echo $?
1
~/projects/rust/mcfly (master)$ true
~/projects/rust/mcfly (master)$ echo $?
0
~/projects/rust/mcfly (master)$ echo $?
0
```

Are you sure it's working for you without McFly @gwk?

**Comment by gwk** [comment]
Yes. It appears that your PROMPT_COMMAND implementation does not restore the exit code, and is relying on the status of `history -cr \$MCFLY_HISTORY`.

Here is a fix; I can open a PR if you like.

```bash
function mcfly_prompt_command {
  local exit_code=$? # Record exit status of previous command.
  history -a $MCFLY_HISTORY # Append history to $MCFLY_HISTORY.
  # Run mcfly with the saved code. It will:
  # * append commands to $HISTFILE, (~/.bash_history by default)
  #   for backwards compatibility and to load in new terminal sessions;
  # * find the text of the last command in $MCFLY_HISTORY and save it to the database.
  mcfly add --exit $exit_code --append-to-histfile
  # Clear the in-memory history and reload it from $MCFLY_HISTORY
  # (to remove instances of '#mcfly: ' from the local session history).
  history -cr $MCFLY_HISTORY
  return $exit_code # Restore the original exit code by returning it.
}

# Set $PROMPT_COMMAND run mcfly_prompt_command and then any existing commands.
PROMPT_COMMAND="mcfly_prompt_command;$PROMPT_COMMAND"
```

**Comment by cantino** [comment]
Your solution makes sense, thank you.

I just wish I understood why I'm not having the issue you're having in my shell.

**Comment by cantino** [comment]
Should be fixed in 9cf7edf3a303f527517efc26ae478dda4090fb2e. Thanks!
